crates/isograph_config/src/compilation_options.rs

```diff
@@ -178,9 +234,23 @@ impl Default for ConfigFileOptionalValidationLevel {
     }
 }
 
+#[derive(Deserialize, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+enum ConfigFileOptionalGenerateFileExtensions {
+    Yes(HashMap<String, String>),
```

Comment:
I've considered two use cases,
- keeping extensions ".tsx" for ".tsx" file
- mapping to javascript ".jsx" for ".tsx" file

I've made it a `HashMap`, so it's possible to use more file extensions, like `.mjs`, `.cjs`, or not standard ones like `.es6` or whatever.

Reply:
When would you need to generate anything besides the name of the actual file?

In addition, if you remap `ts` to `js`, should we also modify `write_artifacts` to generate files with different suffixes?

Also, is the best practice to include extensions, and if so, should we default to including extensions?

Reply:
I'm not sure what best practise here is.

It all depends on tsconfig, and what output user wants, native esm module require explicit ".js" file extensions, I don't know if commonjs allows any extension.

- in case application is transpiled with typescript then it probably runs with `moduleResolution` of `node16` or `nodeNext` where file extensions are required (not sure which .js or .ts are allowed)
- for bundler like vite with `"moduleResolution": "bundler"` imports do not require file extension
- deno has builtin typescript support so it expects `.ts` extensions, and there's an option in typescript for custom file extensions, like `.es6` etc. 



For me most common config is `"moduleResolution": "bundler"` with vite which works without any file extensions. 



I think, theoreticly there should be a different option to customize generated artifacts extension to ".mts" so they can be imported in commonjs modules, not sure if anyone would need this

Reply:
I think there are cases when for `.ts` files `.ts` extension is not allowed, only `.js`

Reply:
Need to check all this to make sure
| File | typescript ESM | typescript CJS | bundler | bundler + `allowImportingTsExtensions` | deno |
|------------------|------------------|------------------|-----------|-----|--|
| .tsx   |  ? |  ?  | `.js` `.jsx` `none` |  `.js` `.jsx` `.tsx` `none`   |?|
| .ts   | ?  |  ?  | `.js` `none` |  `.js` `.ts` `none`   |?|

<details>
  <summary>bundler</summary>

   ```tsx
  {
    "module": "ESNext", 
    "moduleResolution": "Bundler",
  }		
  ```
</details>

Reply:
Different extensions allow for coexistence of CommonJS and ESM files (`.mjs` `.cjs`), explicit file extensions allow for easier transitions between module systems